Library/src/Multiplayer/ReplicatedValue.cpp

```diff
@@ -192,6 +212,66 @@ bool ReplicatedValue::operator!=(const ReplicatedValue& OtherValue) const
 	return (*this == OtherValue) == false;
 }
 
+bool ReplicatedValue::operator<(const ReplicatedValue& OtherValue) const
+{
+	switch (OtherValue.GetReplicatedValueType())
+	{
+		case ReplicatedValueType::Boolean:
+		{
+			return GetBool() < OtherValue.GetBool();
+		}
+		case ReplicatedValueType::Integer:
+		{
+			return GetInt() < OtherValue.GetInt();
+		}
+		case ReplicatedValueType::Float:
+		{
+			return GetFloat() < OtherValue.GetFloat();
+		}
+		case ReplicatedValueType::String:
+		{
+			return GetString() < OtherValue.GetString();
+		}
+		default:
+		{
+			assert(0 && "Unhandled replicated value type!");
+			break;
+		}
+	}
+
+	return false;
+}
+
+bool ReplicatedValue::operator>(const ReplicatedValue& OtherValue) const
+{
+	switch (OtherValue.GetReplicatedValueType())
+	{
+		case ReplicatedValueType::Boolean:
+		{
+			return GetBool() > OtherValue.GetBool();
```

Comment:
I'm genuinely curious if there are cases where checking if bools or greater or lower than each other is useful? can't we do everything we need with equals and  not equals?

Reply:
This is needed to support keys in C++ maps, as they are internally ordered using these operators. However, there won't be many cases where bools will be used as keys.